pkg/statistics/region_collection.go

```diff
@@ -137,8 +157,8 @@ func (r *RegionStatistics) GetOfflineRegionStatsByType(typ RegionStatisticType)
 	r.RLock()
 	defer r.RUnlock()
 	res := make([]*core.RegionInfo, 0, len(r.stats[typ]))
-	for _, r := range r.offlineStats[typ] {
-		res = append(res, r.Clone())
+	for regionID := range r.offlineStats[typ] {
+		res = append(res, r.core.GetRegion(regionID).Clone())
```

Comment:
Will it affect the heartbeat?

Reply:
Considering that it will only be manually called by HTTP API and `pd-ctl`, I don't think it will affect the heartbeat too much.

Reply:
What if we have some stores that are offline and each of them has many regions?

Reply:
I changed this part of the logic to a mixture of old and new. PTAL.